src/sinks/pulsar.rs

```diff
@@ -223,12 +248,19 @@ impl Sink<Event> for PulsarSink {
         Poll::Ready(Ok(()))
     }
 
-    fn start_send(mut self: Pin<&mut Self>, item: Event) -> Result<(), Self::Error> {
+    fn start_send(mut self: Pin<&mut Self>, mut item: Event) -> Result<(), Self::Error> {
         assert!(
             matches!(self.state, PulsarSinkState::Ready(_)),
             "Expected `poll_ready` to be called first."
         );
 
+        let event_time = item
+            .maybe_as_log_mut()
+            .map(|log| log.get(log_schema().timestamp_key())
+                .map(|v| v.as_timestamp().map(|dt| dt.timestamp_millis()))
+                .unwrap_or(None))
+            .unwrap_or(None);
```

Comment:
I think clippy will warn about this, at least it should. There's an alternate form using `.map_or` that avoids the unwrap:
```suggestion
            .map_or(
                None,
                |log| log.get(log_schema().timestamp_key())
                    .map_or(None, |v| v.as_timestamp().map(|dt| dt.timestamp_millis())),
            );
```